platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbCompactionCoordinator.java

```diff
@@ -195,12 +204,22 @@ public void resumeCompaction() throws IOException {
      * All subsequent calls to compacting methods will be ignored until {@link #enableBackgroundCompaction()} is called.
      */
     void stopAndDisableBackgroundCompaction() {
+        compactionEnabled.set(false);
+        // Interrupt all running compaction tasks, if any
         synchronized (compactionFuturesByName) {
             for (var futureEntry : compactionFuturesByName.values()) {
                 futureEntry.cancel(true);
             }
             compactionFuturesByName.clear();
-            compactionEnabled.set(false);
+        }
+        // Wait till all the tasks are stopped
+        final long now = System.currentTimeMillis();
+        try {
+            while ((tasksRunning.get() != 0) && (System.currentTimeMillis() - now < SHUTDOWN_TIMEOUT)) {
+                Thread.sleep(1);
+            }
+        } catch (final InterruptedException e) {
+            logger.warn(MERKLE_DB.getMarker(), "Interrupted while waiting for compaction tasks to complete", e);
         }
```

Comment:
We should also check the main condition (the num of tasks running) after the loop finishes and log some diagnostics (or maybe error out) if it's still not satisfied after the timeout.

Reply:
OK, let me add a log message, if some tasks are still running after the loop

Reply:
Just to be super clear, in this case you want to add it after the try/catch block, so that the swallowed exception doesn't get in the way of the diagnostics.